build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/GenerateWixFile.cs

```diff
@@ -0,0 +1,188 @@
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	/// <summary>
+	/// Generates a .wix file for the contents of ~/android-toolchain/dotnet/packs
+	/// The .wix file can be used to generate the .msi installer for Windows.
+	/// </summary>
+	public class GenerateWixFile : Task
+	{
+		[Required]
+		public string Template { get; set; }
+
+		[Required]
+		public string DestinationFile { get; set; }
+
+		[Required]
+		public string DotNetPath { get; set; }
+
+		[Required]
+		public string DotNetVersion { get; set; }
+
+		[Required]
+		public string MSIVersion { get; set; }
+
+		public override bool Execute ()
+		{
+			var settings = new XmlWriterSettings {
+				OmitXmlDeclaration = true,
+				Indent = true,
+			};
+
+			var directories = new StringBuilder ();
+			var components = new StringBuilder ();
+			using (var packWriter = XmlWriter.Create (directories, settings))
+			using (var componentWriter = XmlWriter.Create (components, settings)) {
```

Comment:
perf-vs-maintenance question: is performance *so* important for something in `build-tools/*` that we need to use `XmlWriter` over `XLinq`?  Yes, XLinq is *slower*, but it's certainly easier to *read* and *maintain*, and it's not like this is part of the customer build…

Reply:
I think it is definitely better to use System.Xml.Linq to _query_ a document, hence Linq in the name.

I'm not sure either is that better/worse for creating an XML file:
```csharp
var components = new XElement ("ComponentGroup",
	new XAttribute ("Id", "ProductComponents"),

	new XElement ("ComponentRef",
		new XAttribute ("Id", "EnableWorkloadResolver")
	)
);
components.Save (...);
```
I get a bit lost on what is an attribute vs element, because they are nested at the same level.

vs:
```csharp
using (var components = XmlWriter.Create (...)) {
	components.WriteStartElement ("ComponentGroup");
	components.WriteAttributeString ("Id", "ProductComponents");
	components.WriteStartElement ("ComponentRef");
	components.WriteAttributeString ("Id", "EnableWorkloadResolver");
	components.WriteEndDocument (); // Or call WriteEndElement twice
}
```

If it's not performance critical, I don't think it's worth rewriting one way or the other.